packages/wallets/src/components/ModalProvider/ModalProvider.tsx

```diff
@@ -3,22 +3,22 @@ import { createPortal } from 'react-dom';
 import { useOnClickOutside } from 'usehooks-ts';
 import { MT5AccountType } from '../../features/cfd/screens';
 import useDevice from '../../hooks/useDevice';
+import { TPlatforms, TMarketTypes } from '../../types';
+
+type TModalState = {
+    marketType?: TMarketTypes.All;
+    platform?: TPlatforms.All;
+};
 
 type TModalContext = {
+    getModalState: <T extends keyof TModalState>(key: T) => TModalState[T];
     hide: () => void;
     isOpen: boolean;
-    modalState?: TModalState;
-    setModalState: (newModalState: Partial<TModalState>) => void;
+    modalState?: Map<keyof TModalState, TModalState[keyof TModalState]>;
+    setModalState: <T extends keyof TModalState>(key: T, value: TModalState[T]) => void;
```

Comment:
Refactored modalState to be a state instead of ref. We want to re-render when user selects `marketType` and sets it into the modal state for instance